libafl/src/events/simple.rs

```diff
@@ -286,23 +273,21 @@ where
 
     // Handle arriving events in the client
     #[allow(clippy::needless_pass_by_value, clippy::unused_self)]
-    fn handle_in_client<S: Stoppable>(
+    fn handle_in_client(
         &mut self,
         state: &mut S,
-        event: Event<I>,
+        event: Event<<S::Corpus as Corpus>::Input>,
     ) -> Result<(), Error>
     where
-        HT: CustomBufHandlerTuple<S>,
-        I: Debug,
+        S: Stoppable,
     {
         match event {
-            Event::CustomBuf { buf, tag } => self.handlers.handle_all(state, &tag, &buf),
             Event::Stop => {
                 state.request_stop();
                 Ok(())
             }
             _ => Err(Error::unknown(format!(
-                "Received illegal message that message should not have arrived: {event:?}."
+                "Received illegal message that message should not have arrived"
```

Comment:
We lose a bit of debugging here. It's probably okay to specify `<S::Corpus as Corpus>::Input: Debug`.